src/ServiceInsight/MessageHeaders/MessageHeadersViewModel.cs

```diff
@@ -29,13 +28,9 @@ public void Handle(SelectedMessageChanged @event)
 
             var headers = storedMessage.Headers;
 
-            using (KeyValues.SuppressChangeNotifications())
+            foreach (var item in headers)
             {
-                KeyValues.AddRange(headers.Select(h => new MessageHeaderKeyValue
-                {
-                    Key = h.Key,
-                    Value = h.Value
-                }));
+                KeyValues.Add(new MessageHeaderKeyValue { Key = item.Key, Value = item.Value });
```

Comment:
Does it matter that we are no longer batching these changes up into a single change notification?